sbt-plugin/src/main/scala/org/scalajs/sbtplugin/ScalaJSPluginInternal.scala

```diff
@@ -742,12 +750,41 @@ object ScalaJSPluginInternal {
 
   // These settings will be filtered by the stage dummy tasks
   val scalaJSRunSettings = Seq(
+      scalaJSMainModuleInitializer := {
+        if (scalaJSUseMainModuleInitializer.value) {
+          mainClass.value.fold {
+            throw new MessageOnlyException(
+                "No main class class detected, but " +
+                "scalaJSUseMainModuleInitializer was set to true. If you do " +
+                "not want to derive a main module initializer from " +
+                "discovered main classes, add " +
+                "`scalaJSMainModuleInitializer := None`")
+          } { mainClass =>
+            Some(ModuleInitializer.mainMethod(mainClass, "main"))
+          }
+        } else {
+          None
+        }
```

Comment:
I almost agreed. But then I realized that your proposal does not support the use case where someone has a custom, non `mainClass`-based module initializer (e.g., their main *method* is not called `main`), but still wants to `run` (hence requires `scalaJSUseMainModuleInitializer`).